Palaso/Reporting/ErrorReport.cs

```diff
@@ -416,24 +419,69 @@ public static string GetOperatingSystemLabel()
 						return String.Format("{0} {1} {2}", si, sr, sc);
 					}
 				}
-				catch(Exception)
-				{ /*lsb_release should work on all supported versions but fall back to the OSVersion.VersionString */ }
+				catch (Exception)
+				{
+					// lsb_release should work on all supported versions but fall back to the OSVersion.VersionString
+				}
 			}
 			else
 			{
-			var list = new List<Version>();
-			list.Add(new Version(System.PlatformID.Win32NT,0,5, "Windows 2000"));
-			list.Add(new Version(System.PlatformID.Win32NT, 1, 5, "Windows XP"));
-			list.Add(new Version(System.PlatformID.Win32NT, 0, 6, "Vista"));
-			list.Add(new Version(System.PlatformID.Win32NT, 1, 6, "Windows 7"));
-			list.Add(new Version(System.PlatformID.Win32NT, 2, 6, "Windows 8"));
-			foreach (var version in list)
-			{
-				if(version.Match(System.Environment.OSVersion))
-					return version.Label + " " + Environment.OSVersion.ServicePack;
+				// https://msdn.microsoft.com/en-us/library/windows/desktop/ms724832%28v=vs.85%29.aspx
+				var list = new List<Version>();
+				list.Add(new Version(PlatformID.Win32NT, 5, 0, "Windows 2000"));
+				list.Add(new Version(PlatformID.Win32NT, 5, 1, "Windows XP"));
+				list.Add(new Version(PlatformID.Win32NT, 6, 0, "Vista"));
+				list.Add(new Version(PlatformID.Win32NT, 6, 1, "Windows 7"));
+				list.Add(new Version(PlatformID.Win32NT, 6, 2, "Windows 8"));
+
+				// Note: Windows might not report Windows 8.1 or 10 unless the app has
```

Comment:
On Windows nowadays, we need to do something like this, instead:

var osClass = new System.Management.ManagementClass("Win32_OperatingSystem");
var oSProperties = new Dictionary<string, string>();
foreach(var queryObj in osClass.GetInstances())
{
	foreach(var prop in queryObj.Properties)
	{
		if(prop.Value!=null)
			oSProperties.Add(prop.Name,prop.Value.ToString());
    }
}

now we have oSProperties["Version"] && oSProperties["Caption"], and this will work even for new versions that you didn't know about when you made the manifest for the installed version of the app.